test/tests/api/v2_0/os_install_tests.py

```diff
@@ -46,8 +50,75 @@ def teardown(self):
     def __get_data(self):
         return loads(self.__client.last_response.data)
 
+    def __get_compute_nodes(self):
+        Api().nodes_get_all()
+        nodes = self.__get_data()
+        compute_nodes = []
+        for n in nodes:
+            type = n.get('type')
+            if type == 'compute':
+                compute_nodes.append(n)
+        LOG.info('compute nodes count {0}'.format(len(compute_nodes)))
+        return compute_nodes
+
+    def __wait_for_completion(self, node, graph_name, graph_instance):
+        id = node.get('id')
+        start_time = date.datetime.now()
+        current_time = date.datetime.now()
+        while True:
+            if (current_time - start_time).total_seconds() > DEFAULT_TIMEOUT_SEC:
+                raise Exception('Timed out after {0} seconds'.format(DEFAULT_TIMEOUT_SEC))
+                break
+            Api().workflows_get()
+            workflows = loads(self.__client.last_response.data)
+            for w in workflows:
+                # LOG.info('print w : {0}'.format(w))
+                if (w.get('node') == id and w.get('injectableName') == graph_name and
+                        w.get('instanceId') == graph_instance):
+                    status = w.get('status')
+                    # LOG.info('{0} - target: {1}, status: {2}'.format(w.get('injectableName'), id, status))
+                    if status == 'succeeded' or status == 'failed' or status == 'canceled':
+                        LOG.info('{0} - target: {1}, status: {2}'.format(w.get('injectableName'), id, status))
+                        if status == 'failed' or status == 'canceled':
+                            LOG.Warning(' workflow {0}: {1}').format(status, w)
+                        assert_equal(status, 'succeeded',
+                            message='{0} - target: {1}, status: {2}'.format(w.get('injectableName'), id, status))
+                        return
+            time.sleep(10)
+            current_time = date.datetime.now()
+            # LOG.info('current time {0} vs start time {1}'.format(current_time, start_time))
+
     def __post_workflow(self, graph_name, nodes, body):
-        workflows().post_workflows(graph_name, timeout_sec=DEFAULT_TIMEOUT_SEC, nodes=nodes, data=body)
+        # check if NODE_INDEX is set
+        index = None
+        try:
+            index = int(os.environ['NODE_INDEX'])
+        except:
+            LOG.info('NODE_INDEX env is not set')
+            workflows().post_workflows(graph_name, timeout_sec=DEFAULT_TIMEOUT_SEC, nodes=nodes, data=body)
+            return
+
+        # check if index is in the array range
+        nodes = self.__get_compute_nodes()
```

Comment:
Actually, there is no guarantee for RackHD API that multiple call for the same API will always return data in the same order. 
Without the guarantee, different index may return the same node document and fail the workflow testing.
To ensure the node document order, you could manually sort the `nodes`.

@brianparry please correct me if my above understanding is wrong.